src/worker/search.worker.ts

```
@@ -0,0 +1,42 @@
+}
+
+self.onmessage = async (e: MessageEvent<SearchMessage>) => {
+  const { query, options, pagination } = e.data;
+
+  try {
+    if (!cache) {
+      // تحميل البيانات في المسار الخلفي
+      const [quran, morphology, wordMap] = await Promise.all([
+        loadQuranData(),
+        loadMorphology(),
+        loadWordMap()
```

ESLint

[error] Insert `,`

(prettier/prettier)